src/OnboardingSPA/components/Drawer/DrawerPanel/DesignHeaderMenu.js

```diff
@@ -1,17 +1,139 @@
-import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+
+import HeaderMenuPreview from '../../HeaderMenuPreview';
+import { store as nfdOnboardingStore } from '../../../store';
+import { getPatterns, getHeaderMenuPatterns, getDefaultHeaderMenu } from '../../../utils/api/patterns';
+import { GlobalStylesProvider } from '../../../components/LivePreview';
+
+import {
+	THEME_STATUS_ACTIVE,
+	THEME_STATUS_NOT_ACTIVE,
+} from '../../../../constants';
+
+const DesignHomepageMenu = () => {
+	const headerMenuSlugs = [ 
```

Comment:
Shouldn't we be using this mapping to reduce API Calls and do the computation in Frontend rather than a API Request. Perhaps we can store in the Flow data and have a useEffect inside the Page to keep checking flow data and making a subsequent change in the Preview.
If we are already doing that we can remove this var as I don't see it being used anywhere.